src/robot/dial.ts

```diff
@@ -0,0 +1,122 @@
+import { RobotClient } from './Client';
+
+interface DialDirectConf {
+  authEntity?: string;
+  host: string;
+  locationSecret?: string;
+}
+
+/** Check if a url corresponds to a local connection via heuristic */
+const isLocalConnection = (url: string) => url.includes('.local');
+
+const dialDirect = async (conf: DialDirectConf): Promise<RobotClient> => {
+  // eslint-disable-next-line no-console
+  console.info('dialing via gRPC...');
+
+  if (!isLocalConnection(conf.host)) {
+    throw new Error(
+      `cannot dial "${conf.host}" directly, please use a local url instead.`
+    );
+  }
+
+  const client = new RobotClient(conf.host);
+
+  let creds;
+  if (conf.locationSecret) {
+    creds = {
+      payload: conf.locationSecret,
+      type: 'robot-location-secret',
+    };
+  }
+  await client.connect(conf.authEntity, creds);
+
+  // eslint-disable-next-line no-console
+  console.info('connected via gRPC');
+
+  return client;
+};
+
+interface ICEServer {
+  urls: string;
+  username: string;
+  credential: string;
+}
+
+interface DialWebRTCConf {
+  authEntity?: string;
+  host: string;
+  locationSecret?: string;
+  // WebRTC
+  signalingAddress: string;
+  iceServers: ICEServer[];
+}
+
+const dialWebRTC = async (conf: DialWebRTCConf): Promise<RobotClient> => {
+  // eslint-disable-next-line no-console
+  console.info('dialing via WebRTC...');
+
+  const impliedURL = conf.host;
+  const signalingAddress = conf.signalingAddress;
+  const iceServers = conf.iceServers;
+
+  const rtcConfig = { iceServers };
+  const clientConf = {
+    enabled: true,
+    host: conf.host,
+    signalingAddress,
+    rtcConfig,
+  };
+  const client = new RobotClient(impliedURL, clientConf);
+
+  let creds;
+  if (conf.locationSecret) {
+    creds = {
+      payload: conf.locationSecret,
+      type: 'robot-location-secret',
+    };
+  }
+  await client.connect(impliedURL, creds);
+
+  // eslint-disable-next-line no-console
+  console.info('connected via WebRTC');
+
+  return client;
+};
+
+type Conf = DialDirectConf | DialWebRTCConf;
+
+const isDialWebRTCConf = (value: Conf): value is DialWebRTCConf => {
+  const conf = value as DialWebRTCConf;
+
+  if (typeof conf.signalingAddress !== 'string') return false;
+  if (!(conf.iceServers instanceof Array)) return false;
```

Comment:
ice servers are not strictly necessary to connect

Reply:
adjusted to reflect this (took an embarrassing number of commits...)